### DISCUSSION TITLE
Problem with gravity (blocks)

### DISCUSSION THREAD

**Discussion by ElAndree** [Minestom/Minestom#666] [Help] [answered]
Hey there,

i am currently trying to implement gravity blocks (sand, gravel) and i am orientating at the Minestom VRI, but having some problems.

So first thing is (this is my current code)
```
public void onPlace(Placement placement) {
        Instance instance = placement.getInstance();
        Point position = placement.getBlockPosition();

        instance.scheduleNextTick((instance1 -> checkFall(instance1, position)));
    }

    public void checkFall(Instance instance, Point position) {
        Block block = instance.getBlock(position);
        Block below = instance.getBlock(position.blockX(), position.blockY() - 1, position.blockZ());

        // Exit out now if block below is solid
        if (below.isSolid()) {
            return;
        }
        instance.setBlock(position, Block.AIR);

        Pos initialPosition = new Pos(position.x() + 0.5f, Math.round(position.y()), position.z() + 0.5f);
        Entity fallingBlockEntity = new Entity(EntityType.FALLING_BLOCK);
        ((FallingBlockMeta)fallingBlockEntity.getEntityMeta()).setBlock(baseBlock);

        fallingBlockEntity.setInstance(instance);
        fallingBlockEntity.teleport(initialPosition);
    }
```

Is there a better way to spawn a falling block entity or do i really have to cast the entity meta to a FallingBlockMeta?

And my second, bigger problem: The gravity block is very laggy - do you know why this is so?
I experienced the same with other entities falling.
(see here: https://youtu.be/ligo5--S728 (pls ignore the audio))

Thank you!!

**Comment by emortaldev** [comment, marked as answer]
Yes that is the optimal way to spawn one, however you can remove the unnecessary `fallingBlockEntity.teleport(initialPosition);` line and instead do `fallingBlockEntity.setInstance(instance, initialPosition);`

> The gravity block is very laggy - do you know why this is so?

I experience this problem too

**Reply by TheMode** [reply]
(Prediction issue should be fixed by 2006098ee28f0332480c00fe42d6eb23d0c5dc77)

**Reply by ElAndree** [reply]
> (Prediction issue should be fixed by [2006098](https://github.com/Minestom/Minestom/commit/2006098ee28f0332480c00fe42d6eb23d0c5dc77))

LivingEntity (something like zombie, blaze, etc) is also missing, isnt it?

**Reply by TheMode** [reply]
Not all entity velocity are predicted
